site-src/geps/gep-1619.md

```diff
@@ -0,0 +1,89 @@
+
+### Defining Session Persistence
+
+Session persistence is when a client request is directed to the same backend server for the duration of a "session". It is achieved when a client directly provides information, such as a header, that a proxy uses as a reference to direct traffic to a specific server. Persistence is an exception to load balancing: a persistent client request bypasses the proxy's load balancing algorithm, going directly to a backend server it has previously established a session with.
+
+Some of the concerns of session persistence are the duration and expiration of the session, security of the transaction stream, and storage of the context or state. Application developers use session persistence to improve the performance and reduce overall storage needs by aligning each transaction and its cached data with the same server.
+
+Session affinity, not to be confused with session persistence, uses an existing attribute of the request to consistently send to the same backend. Session affinity can be considered a weaker form of session persistence: it is not guaranteed to persist a connection to the same backend server if certain attributes of the request or the backends are changed.
+
+### Achieving Session Persistence and Session Affinity
+
+Session persistence is achieved with the application layer, while session affinity often uses, but is not limited to, attributes below the application layer.
+
+Here are a some mechanisms to achieving both:
+
+**Cookie-Based Session Persistence**
+
+The most common mechanism is by using cookies (described by [RFC6265](https://www.rfc-editor.org/rfc/rfc6265)) with the set-cookie HTTP response header. A client will use the provided value in the set-cookie response header in a cookie request header in subsequent requests. Proxies can use this cookie header to maintain a persistent connection to a single backend server on behalf of the client.
+
+**Header-Based Session Persistence**
+
+Header-based stateful sessions are achieved by a backend providing an arbitrarily-named HTTP response header and the client using the same arbitrarily-named header in subsequent HTTP requests. Proxies can use this arbitrarily-named header to maintain a persistent connection to a single backend server on behalf of the client.
+
+**Session Affinity**
+
+Session affinity can be achieved by deterministic load balancing algorithms or a proxy feature that tracks IP-to-backend associations such as [HAProxy's stick tables](https://www.haproxy.com/blog/introduction-to-haproxy-stick-tables/) or [Cilium's session affinity](https://docs.cilium.io/en/v1.12/gettingstarted/kubeproxy-free/#id2). 
+
+### Implementations
+To help understand the scope of session persistence and session affinity, we've created a table describing how each implementation achieves it. Input from implementations is appreciated to complete this table.
```

Comment:
It may help to clarify that this table intends to provide the kinds of session persistence and affinity that are supported by each implementation. Just for the sake of maintaining this going forward, it may be easiest to just focus on underlying implementations such as Envoy, NGINX, HAProxy, etc. Even if an implementation based on one of those does not support a feature today, that may just be an omission of their APIs and not a limitation of their underlying dataplane.

Reply:
+Java Servlets please. Persistent sessions since ~2000 encoded in the spec, multiple implementations and support in major proxies.